bin/plot_preproc

```diff
@@ -0,0 +1,169 @@
+
+import numpy as np
+import matplotlib.pyplot as plt
+import fitsio
+
+from astropy.table import Table
+from astropy.visualization import ZScaleInterval
+
+from desiutil.log import get_logger
+from desispec.io import findfile, read_xytraceset
+
+def get_cameras(fiber, wavelength):
+    """
+    Get list of cameras that cover this fiber and wavelength
+
+    Args:
+        fiber (int): fiber number 0-4999
+        wavelength (float): wavelength in Angstroms
+
+    Returns (list of str): cameras that cover that fiber,wavelength
+    """
+    bmin, bmax = 3600, 5800
+    rmin, rmax = 5760, 7620
+    zmin, zmax = 7520, 9824
+
+    spectrograph = fiber//500
+    cameras = list()
+    if bmin <= wavelength <= bmax:
+        cameras.append(f'b{spectrograph}')
+    if rmin <= wavelength <= rmax:
+        cameras.append(f'r{spectrograph}')
+    if zmin <= wavelength <= zmax:
+        cameras.append(f'z{spectrograph}')
+
+    return cameras
+
+def read_thumbnail(night, expid, camera, fiber, wavelength, size=51):
+    """
+    Read thumbnail image for night,expid,camera centered on fiber,wavelength
+
+    Args:
+        night (int): YEARMMDD night
+        expid (int): exposure ID
+        camera (str): spectrograph camera, e.g. b0, r1, z9
+        fiber (int): fiber number 0-4999
+        wavelength (float): wavelength in Angstrom
+
+    Optional:
+        size (int): size of thumbnail in pixels
+
+    Returns: (thumbnail_image, thumbnail_mask)
+    """
+    psffile = findfile('psf', night, expid, camera, readonly=True)
+    xy = read_xytraceset(psffile)
+
+    x = int(xy.x_vs_wave(fiber%500, wavelength))
+    y = int(xy.y_vs_wave(fiber%500, wavelength))
+
+    preprocfile = findfile('preproc', night, expid, camera, readonly=True)
+    with fitsio.FITS(preprocfile) as fx:
+        img = fx['IMAGE'].read()
+        mask = fx['MASK'].read()
+
+    xyslice = (slice(y-size//2, y+size//2+1), slice(x-size//2, x+size//2+1))
+    return img[xyslice], mask[xyslice]
+
+def plot_thumbnail(img, mask):
+    """
+    Plot thumbnail imgage with overlaid transparent red mask
+    """
+    zscale = ZScaleInterval()
+    vmin, vmax = zscale.get_limits(img)
+    plt.imshow(img, vmin=vmin, vmax=vmax)
+
+    # RGBA overlay for mask in red, unmasked is transparent
+    overlay = np.zeros((*mask.shape, 4), dtype=float)
+    overlay[mask!=0, 0] = 1.0  # red
+    overlay[mask!=0, 3] = 0.5  # transparency
+    plt.imshow(overlay)
+
+def plot_crosshair(x,y):
+    """
+    Plot magenta crosshair at x,y
+    """
+    fmt = 'm-'
+    a,b = 2,5
+    plt.plot([x-b,x-a], [y,y], fmt)
+    plt.plot([x+a,x+b], [y,y], fmt)
+    plt.plot([x,x], [y-b,y-a], fmt)
+    plt.plot([x,x], [y+a,y+b], fmt)
+
+def main():
+    p = argparse.ArgumentParser()
+    p.add_argument('-f', '--fiber', type=int, required=True, help="fiber number")
+    p.add_argument('-w', '--wavelength', type=float, required=True, help="wavelength in Angstrom")
+    p.add_argument('-t', '--tileid', type=int, help="Tile ID")
```

Comment:
It would be nice to also accommodate --expid for either calibration exposures or if the user know what exposure they want to look at.

Reply:
Thanks, I think that is a perfectly reasonable constraint for calibration images. If a person knows the EXPID for a cal, hopefully they also have information on what night is was acquired on, or they can do their own search.